module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountActionsTest.php

```diff
@@ -214,13 +218,127 @@ public function testChangeEmail()
+                    ]
+                ],
+                '',
+                ['', 'A', 'B', 'C'],
+                'username3'
+            ],
+        ];
+    }
+
+    /**
+     * Test default pick up location
+     *
+     * @dataProvider getTestDefaultPickUpLocationData
+     *
+     * @return void
+     */
+    public function testDefaultPickUpLocation(
+        array $demoOverrides,
+        string $expectedDefaultPickUp,
+        array $expectedPickUpChoices,
+        string $username
+    ) {
+        // Setup config
+        $this->changeConfigs(
+            [
+                'Demo' => $demoOverrides,
+                'config' => [
+                    'Catalog' => ['driver' => 'Demo'],
+                ]
+            ]
+        );
+
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl());
+        $page = $session->getPage();
+
+        // Create account
+        $this->clickCss($page, '#loginOptions a');
+        $this->clickCss($page, '.modal-body .createAccountLink');
+        $this->fillInAccountForm(
+            $page,
+            [
+                'username' => $username,
+                'email' => "$username@ignore.com"
+            ]
+        );
+        $this->clickCss($page, '.modal-body .btn.btn-primary');
+
+        // Go to profile page:
+        $this->waitForPageLoad($page);
+        $session->visit($this->getVuFindUrl('/MyResearch/Profile'));
+
+        // Do patron login:
+        $this->submitCatalogLoginForm($page, 'catuser', 'catpass');
+
+        // Check the default library and possible values:
+        $this->assertEquals(
+            $expectedDefaultPickUp,
+            $this->findCss($page, '#home_library')->getValue()
+        );
+        foreach ($expectedPickUpChoices as $i => $expected) {
+            $this->assertEquals(
+                $expected,
+                $this->findCss($page, '#home_library option', null, $i)->getValue()
+            );
+        }
+        // Make sure there are no more pick up locations:
+        $this->unFindCss(
+            $page,
+            '#home_library option',
+            null,
+            count($expectedPickUpChoices)
+        );
+
+        // Change the default and verify:
+        $this->findCss($page, '#home_library')->setValue('B');
+        $this->clickCss($page, '#profile_form .btn');
+        $this->waitForPageLoad($page);
+        $this->assertEquals('B', $this->findCss($page, '#home_library')->getValue());
+
+        // If emptying the selection is available, test it:
+        if (in_array('', $expectedPickUpChoices)) {
+            $this->findCss($page, '#home_library')->setValue('');
+            $this->clickCss($page, '#profile_form .btn');
+            $this->waitForPageLoad($page);
+            $this->assertEquals(
+                '',
+                $this->findCss($page, '#home_library')->getValue()
+            );
+            echo "CHECKED!\n";
```

Comment:
I don't think you want this echo statement.

Reply:
Oops. Removed.